helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java

```diff
@@ -356,10 +532,34 @@ private void addStoppableCheck(Map<String, StoppableCheck> stoppableChecks, Stri
     return instancesForNextCheck;
   }
 
+  private boolean isNonBlockingCheck(StoppableCheck stoppableCheck) {
```

Comment:
Why we do the validation and filterout the nonblocking checks? Why let's just filter out the verification rule before check perform?

Reply:
Maybe user want to know if the check pass or not..? Like how health check does it..?

Reply:
Health check is something we dont want to check. From performance, better we skip those user defined checks for easy use.

Reply:
Had an offline discuss. We allow user to select what health check to perform.  Currently user can choose (HelixHealthChekc and/or CustomeHealthCheck). User can choose finer grain health check like (HELIX:INSTANCE_NOT_ENABLED, CUSTOM_PARTITION_HEALTH_FAILURE:PARTITION_INITIAL_STATE_FAIL etc). 
And we keep the continousOnFailure key word to be consistent as stoppable API. 
An TODO is added for this.